docs/building.md

```diff
@@ -0,0 +1,176 @@
+- [Platform Owner](https://github.com/fobnail/fobnail-platform-owner) (C)
+- [Attester](https://github.com/fobnail/fobnail-attester) (C)
+
+There is also [Fobnail SDK](https://github.com/fobnail/fobnail-sdk) that
+simplifies building of firmware by using Docker container, but only
+[one script](https://github.com/fobnail/fobnail-sdk/blob/main/run-fobnail-sdk.sh)
+from this repository is actually needed.
+
+Following steps were successfully performed on Ubuntu 22.04.1 LTS.
+
+## Fobnail Token firmware
+
+Start by installing prerequisites and adding current user to `docker` group:
+
+```shell
+sudo apt -y install git docker.io
+sudo usermod -aG docker $USER
+```
+
+Log out of the desktop session then log in again - this is required after making
+the change to user groups. You may confirm it by running `groups`.
+
+Clone the repository along with its submodules:
+
+```shell
+git clone https://github.com/fobnail/fobnail --recurse-submodules
+```
+
+Build process for Token firmware is by far the most complicated one. For that
+reason Fobnail SDK is strongly suggested. To use it, simply download script
+`run-fobnail-sdk.sh` either by cloning whole repository or just this one script
+and save it to a directory in your `PATH` environment variable:
+
+```shell
+sudo wget -O /usr/local/bin/run-fobnail-sdk.sh \
+  https://raw.githubusercontent.com/fobnail/fobnail-sdk/main/run-fobnail-sdk.sh
+sudo chmod +x /usr/local/bin/run-fobnail-sdk.sh
+```
+
+With the script installed, one can prepare for the build process itself.
+Firmware can be build for a physical Fobnail Token or simulated on PC, with
+different preparatory steps required. In both cases, first execution of
+`build.sh` downloads Docker container image, which can be both time- and
+bandwidth-consuming.
+
+#### Environment variables common for both targets
+
+Fobnail Token firmware is configured with environment variables passed to
+`build.sh`.
+
+Note that due to the way Docker mounts directories all files and directories
+pointed to by following variables must be located somewhere in `fobnail`
+directory.
+
+- `FOBNAIL_PO_ROOT` - **required** option, must point to valid PEM or DER file
+  with Platform Owner's root certificate. See [this document](/keys_and_certificates/#platform-owner-certificate-chain)
+  for description of PO certificate chain and instructions for building such.
+
+- `FOBNAIL_EK_ROOT_DIR` - points to directory with TPM root certificates.
+  Fobnail repository includes [such directory](https://github.com/fobnail/fobnail/tree/main/tpm_ek_roots)
+  which normally should be used as `FOBNAIL_EK_ROOT_DIR`, but you may change it
+  if certificates supplied in repository became outdated. Either this or
+  `FOBNAIL_EXTRA_EK_ROOT` (or both) **must be specified**, but build process
+  won't fail when neither is passed.
+
+- `FOBNAIL_EXTRA_EK_ROOT` - points to one specific TPM certificate. Useful when
+  testing with TPM emulator, or to limit Fobnail usage to one specific TPM
```

Comment:
Do we have documentation how to test with emulator?

Reply:
Added, note that it requires https://github.com/fobnail/fobnail-attester/pull/42 to be fully compatible with instructions.

Reply:
> Not yet, it was partially covered in one of previous blog posts.

I think we should avoid mentioning use cases that none can validate now.


> Added, note that it requires https://github.com/fobnail/fobnail-attester/pull/42 to be fully compatible with instructions.

I'm sorry, but I don't understand how it is related to emulator use case.

Reply:
Instructions are in `tpm-simulators.md` on this branch, PR linked above must be applied first or certificate created with the script won't work.

Reply:
Ok, I will get back to that when I will have time for testing.